src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Yarp.ReverseProxy.Forwarder;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    internal class DefaultHttpProxyService : IHttpProxyService
+    {
+        private IHttpForwarder _httpForwarder;
+        private HttpMessageInvoker _messageInvoker;
+        private ForwarderRequestConfig _forwarderRequestConfig;
+        private string _proxyEndpoint;
+
+        public DefaultHttpProxyService(IHttpForwarder httpForwarder)
+        {
+            _httpForwarder = httpForwarder;
+
+            var handler = new SocketsHttpHandler();
+            _messageInvoker = new HttpMessageInvoker(handler);
+            _forwarderRequestConfig = new ForwarderRequestConfig();
+
+            // TODO: Update this logic. Port should come through configuration.
+            var port = Environment.GetEnvironmentVariable("AZURE_FUNCTIONS_HTTP_PROXY_PORT") ?? "5555";
```

Comment:
Should we read this from `IConfiguraiton` / `IOptions<ConfigForThisType>` ? Do we have a configuration source which we populate from the host.json? That allows user to override this value via host.json if needed?

Reply:
https://github.com/Azure/azure-functions-dotnet-worker/issues/1419 - wip, this implementation is for prototyping purposes.